tests/Integration/Presentation/Presenters/MembershipApplicationConfirmationHtmlPresenterTest.php

```diff
@@ -0,0 +1,72 @@
+<?php
+
+namespace WMDE\Fundraising\Frontend\Tests\Integration\Presentation\Presenters;
+
+use WMDE\Fundraising\Frontend\MembershipContext\Tests\Data\ValidMembershipApplication;
+use WMDE\Fundraising\Frontend\MembershipContext\UseCases\ShowMembershipApplicationConfirmation\ShowMembershipAppConfirmationResponse;
+use WMDE\Fundraising\Frontend\Presentation\Presenters\MembershipApplicationConfirmationHtmlPresenter;
+use WMDE\Fundraising\Frontend\Presentation\TwigTemplate;
+
+/**
+ * @covers WMDE\Fundraising\Frontend\Presentation\Presenters\MembershipApplicationConfirmationHtmlPresenter
+ *
+ * @licence GNU GPL v2+
+ * @author Kai Nissen < kai.nissen@wikimedia.de >
+ */
+class MembershipApplicationConfirmationHtmlPresenterTest extends \PHPUnit_Framework_TestCase {
+
+	private const STATUS_BOOKED = 'status-booked';
+	private const STATUS_UNCONFIRMED = 'status-unconfirmed';
+
+	/** @dataProvider applicationStatusProvider */
+	public function testWhenPresenterPresents_itPassesParametersToTemplate( $isConfirmed, $expectedMappedStatus ) {
+		$twig = $this->getMockBuilder( TwigTemplate::class )->disableOriginalConstructor()->getMock();
+		$twig->expects( $this->once() )
+			->method( 'render' )
+			->with( $this->getExpectedRenderParams( $expectedMappedStatus ) );
+
+		$membershipApplication = ValidMembershipApplication::newDomainEntityUsingPayPal();
+		if ( $isConfirmed === true ) {
+			$membershipApplication->confirm();
+		}
```

Comment:
This seems like abusing data providers a bit. Having two tests with factored out Arrange Act and Assert parts as appropriate is probably nicer.